mediaire_toolbox/transaction_db/transaction_db.py

```diff
@@ -137,22 +148,37 @@ def set_failed(self, id_: int, cause: str):
             self.session.rollback()
             raise
 
-    def set_completed(self, id_: int, clear_error: bool = True):
+    def set_completed(self, id_: int, clear_error: bool=True):
         """to be called when the transaction completes successfully.
         Error field will be set to '' only if clear_error = True.
-        End_date automatically adjusted."""
+        End_date automatically adjusted. Status is automatically set to 
+        'unseen'."""
         try:
             t = self._get_transaction_or_raise_exception(id_)
             t.task_state = TaskState.completed
+            t.status = 'unseen'
             t.end_date = datetime.datetime.utcnow()
             if clear_error:
                 t.error = ''
             self.session.commit()
         except:
             self.session.rollback()
             raise
+        
+    def set_status(self, id_: int, status: str):
+        """to be called e.g. when the radiologist visits the results of a study
+        in the new platform ('reviewed') or the report is sent to the PACS
+        ('sent_to_pacs') ..."""
+        try:
+            t = self._get_transaction_or_raise_exception(id_)
+            t.status = status
```

Comment:
perhaps we should raise an error when the given status is not `sent_to_pacs` or `reviewed` to avoid typos?

Reply:
I wanted to leave this method open to any status, in case we need more in the future.